DDD Workshop.ipynb

```diff
@@ -0,0 +1,547 @@
+    "\n",
+    "`AIRPORT,SEATS,OCCUPIED,TIME`\n",
+    "\n",
+    "`LAX,200,180,WEEKDAY`\n",
+    "\n",
+    "`JFK,200,161,WEEKDAY`\n",
+    "\n",
+    "`LAX,200,186,WEEKDAY`\n",
+    "\n",
+    "`YVR,200,0.8315980689983522,WEEKDAY`\n",
+    "\n",
+    "`LAX,250,227,REDEYE`\n",
+    "\n",
+    "`JFK,250,226,REDEYE`\n",
+    "\n",
+    "`LAX,250,228,REDEYE`\n",
+    "\n",
+    "`YVR,250,0.8663797750002538,REDEYE`\n",
+    "\n",
+    "&vellip;\n",
+    "\n",
+    "We will access the full 1800 line file in the code, but all of the contents will follow the format above.\n",
+    "\n",
+    "\n",
+    "## Types of Conclusions\n",
+    "\n",
+    "#### What makes a backward-looking conclusion?\n",
+    "\n",
+    "These types of conclusions are for very broad generalizations and are mostly observations, rather than predictions. There is a way to categorize the types of insights gained from data analysis: Descriptive, Diagnostic, Predictive, and Prescriptive. These can be thought of as increasing both specificities and work required: \n",
+    "\n",
+    "![](https://i.imgur.com/HlYuOwH.png)\n",
+    "\n",
+    "<!-- cite: https://www.scnsoft.com/blog/4-types-of-data-analytics -->\n",
+    "\n",
+    "A backward-looking conclusion is then one of the first two of these and is broader in scope. For example, in looking at the flight data we see that there are twice as many flights to LA relative to Vancouver or New York. This information is probably easy to see, and we can confirm it over the entire dataset.\n",
+    "\n",
+    "###### Python Code:\n",
+    "\n",
+    "_The code below is some vanilla python3, a programming language very popular with data analysts. We'll be working out of a text file named `flightdata.csv`, which is the raw data you saw above, and eventually storing the cleaned data in another file named `cleanedflightdata.csv`, which has the fixed values for Vancouver. See below:_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'LAX': 898, 'YVR': 449, 'JFK': 449}\n"
+     ]
+    }
+   ],
+   "source": [
+    "data_file = open(\"flight_data.csv\", \"r\")\n",
+    "\n",
+    "counts = {\"LAX\": 0, \"YVR\": 0, \"JFK\": 0}\n",
+    "for line in data_file:\n",
+    "    airport = line[:3]\n",
+    "    if airport in counts.keys():\n",
+    "        counts[airport] += 1\n",
+    "\n",
+    "data_file.close()\n",
+    "\n",
+    "print(counts)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Other similar backward-looking conclusions might be is the size of the planes is related to the destinations, is the occupancy is related to the plane sizes alone, etc. Most backward-looking conclusions can be drawn from the data directly, provided you are aware of the potential fallacies and have verified your data.\n",
```

Comment:
Reword to: Other similar backward-looking conclusions might include the size of the planes related to the destinations, the occupancy related to the plane sizes alone, etc. Most backward-looking conclusions can be drawn from the data directly, provided you are aware of the potential fallacies and have verified your data.